src/dbnode/client/host_queue.go

```diff
@@ -950,6 +955,28 @@ func (q *queue) asyncTruncate(op *truncateOp) {
 	})
 }
 
+var noopCancel = context.CancelFunc(func() {})
+
+func (q *queue) thriftContextRead(
+	callingContext context.Context,
+) (thrift.Context, context.CancelFunc) {
+	var (
+		timeout = q.opts.FetchRequestTimeout()
+		cancel  = noopCancel
+	)
+	if callingContext != nil {
+		// Use the original context for the call.
+		_, ok := callingContext.Deadline()
```

Comment:
we should enforce a deadline is set and it meets some minimum threshold ( > 1s ?) before sending to the db

Reply:
Yup, done.